blackmarket/Managers/SessionManager.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  SessionDataManager.swift
+//  ios-base
+//
+//  Created by Juan Pablo Mazza on 11/8/16.
+//  Copyright © 2016 Rootstrap Inc. All rights reserved.
+//
+
+import UIKit
+import Alamofire
+import Combine
+
+internal class SessionManager: CurrentUserSessionProvider {
+    
+    static let Authenticated = PassthroughSubject<Bool, Never>()
+    
+    static func IsAuthenticated() -> Bool {
+        return SessionManager.shared.validSession
+    }
+    
+    static let shared = SessionManager()
+    
+    private let userDefaults: UserDefaults
+    
+    init(userDefaults: UserDefaults = .standard) {
+        self.userDefaults = userDefaults
+    }
+    
+    var currentSession: Session? {
+        get {
+            if
+                let data = userDefaults.data(forKey: "ios-base-session"),
```

Comment:
please use a different key for the user defaults value. It is also a good idea to move it to a constant so you change it only in one place.